src/index.js

```diff
@@ -32,37 +32,39 @@ class RetryChunkLoadPlugin {
             this.options.chunks.includes(currentChunkName);
           if (!addRetryCode) return source;
           const script = `
-          var oldGetScript = ${RuntimeGlobals.getChunkScriptFilename};
-          var oldLoadScript = ${RuntimeGlobals.ensureChunk};
-          var queryMap = new Map();
-          var countMap = new Map();
-          var maxRetries = ${maxRetries};
-          ${RuntimeGlobals.getChunkScriptFilename} = function(chunkId){
-            var result = oldGetScript(chunkId);
-            return result + (queryMap.has(chunkId) ? '?' + queryMap.get(chunkId)  : '');
-          };
-          ${RuntimeGlobals.ensureChunk} = function(chunkId){
-            var result = oldLoadScript(chunkId);
-            return result.catch(function(error){
-              var retries = countMap.has(chunkId) ? countMap.get(chunkId) : ${maxRetries};
-              if (retries < 1) {
-                var realSrc = oldGetScript(chunkId);
-                error.message = 'Loading chunk ' + chunkId + ' failed after ${maxRetries} retries.\\n(' + realSrc + ')';
-                error.request = realSrc;${
-                  this.options.lastResortScript
-                    ? this.options.lastResortScript
-                    : ''
+          if(typeof ${RuntimeGlobals.require} !== "undefined") {
```

Comment:
Is it possible check on compiling?

Reply:
I also looked at checking if the mainTemplate.require hook is called first, and register this hook only when that one has been called. I chose this route because (1) i dont know how that approach will interact with any other plugins touching these hooks and (2) the mainTemplate.require hook is slated to be removed in WP6 (per TODO comments in webpack source)